### ISSUE TITLE
[bug][postgres] For custom SQL functions returning one row server generate wrong graphql type.

### ISSUE THREAD

**Issue by asan** [hasura/graphql-engine#7318]
For custom SQL functions returning one row server generate array graphql type, but returns only one record.

<img width="1352" alt="Screen Shot 2021-07-28 at 22 16 33" src="https://user-images.githubusercontent.com/154874/127383640-62884935-4450-450e-85f3-ace2da9d68b7.png">

Steps to reproduce:

Insert SQL with "track it" checkbox checked:

```sql
CREATE TABLE test (
    id integer primary key,
    name text
);

INSERT INTO test (id, name) VALUES (1, 'omg');
INSERT INTO test (id, name) VALUES (2, 'test');

CREATE FUNCTION test_by_name(name_t text) RETURNS test LANGUAGE sql STABLE AS $$
SELECT
  *
FROM
  test
WHERE
  name = name_t 
LIMIT 1
$$
```

Execute query
```
query MyQuery {
  test_by_name(args: {name_t: "test"}) {
    name
  }
}
```
result is single object:
```
{
  "data": {
    "test_by_name": {
      "name": "test"
    }
  }
}
```
But type in "Documentation Explorer" described as array `[test!]!`
```
test_by_name(
args: test_by_name_args!
distinct_on: [test_select_column!]
limit: Int
offset: Int
order_by: [test_order_by!]
where: test_bool_exp
): [test!]!
```

Plus function has many excessive arguments for object relationship like limit and offset.

**Comment by tirumaraiselvan** [comment]
Dupe of https://github.com/hasura/graphql-engine/issues/7109

**Comment by asan** [comment]
my bad, hope reproduction steps will help

**Comment by asan** [comment]
fixed in 2.0.7
